src/chat.rs

```diff
@@ -1586,6 +1597,56 @@ pub fn set_gossiped_timestamp(context: &Context, chat_id: u32, timestamp: i64) {
     }
 }
 
+pub fn shall_attach_selfavatar(context: &Context, chat_id: u32) -> Result<bool, Error> {
+    // versions before 12/2019 already allowed to set selfavatar, however, it was never sent to others.
+    // to avoid sending out previously set selfavatars unexpectedly we added this additional check.
+    // it can be removed after some time.
+    if !context
+        .sql
+        .get_raw_config_bool(context, "attach_selfavatar")
+    {
+        return Ok(false);
+    }
+
+    let resend_every_days = 14;
+    let timestamp_some_days_ago = time() - resend_every_days * 24 * 60 * 60;
+    let needs_attach = context.sql.query_map(
+        "SELECT c.selfavatar_sent
+           FROM chats_contacts cc
+           LEFT JOIN contacts c ON c.id=cc.contact_id
+          WHERE cc.chat_id=? AND cc.contact_id!=?;",
+        params![chat_id, DC_CONTACT_ID_SELF],
+        |row| Ok(row.get::<_, i64>(0)),
+        |rows| {
+            let mut needs_attach = false;
+            for row in rows {
+                if let Ok(selfavatar_sent) = row {
+                    let selfavatar_sent = selfavatar_sent?;
+                    if selfavatar_sent < timestamp_some_days_ago {
```

Comment:
it is fine and safe to do `if selfavatar_sent? < timestamp_some_days_ago {` but maybe it looks a bit weird to C-devs thinking of ?" as a ternary operator :)